packages/newspringchurchapp/src/prayer/PrayerList/index.js

```diff
@@ -1,210 +1,239 @@
-   */
-
-  scrollToNext = () => {
-    this.scroller.scrollToIndex({
-      index: this.state.currentCardIndex + 1,
-      animated: true,
-    });
-    this.setState((prevState) => ({
-      currentCardIndex: prevState.currentCardIndex + 1,
-    }));
+  static defaultProps = {
+    query: 'prayers',
   };
 
+  // TODO: this may be better pulled from a content channel???
   calculateQuery = () => {
-    const { navigation } = this.props;
-    const list = navigation.getParam('list', '');
-
-    let query;
-    let queryName;
-    let type;
+    let queryTag;
+    let label;
     let variables;
 
-    switch (list) {
-      case 'GroupPrayerList':
-        query = GET_GROUP_PRAYERS;
-        queryName = 'groupPrayers';
-        type = 'community';
-        break;
-      case 'ChurchPrayerList':
-        query = GET_PRAYERS;
-        queryName = 'prayers';
-        type = 'church';
+    switch (this.props.query) {
+      case 'groups':
+        queryTag = GET_GROUP_PRAYERS;
+        label = 'My Community';
         break;
-      case 'CampusPrayerList': {
+      case 'campus': {
         const {
           currentUser: { profile: { campus: { id } = {} } = {} } = {},
         } = cache.readQuery({
           query: getUserProfile,
         });
-        query = GET_CAMPUS_PRAYERS;
-        queryName = 'campusPrayers';
-        type = 'campus';
+        queryTag = GET_CAMPUS_PRAYERS;
+        label = 'My Campus';
         variables = { campusId: id };
         break;
       }
       default:
-        query = GET_PRAYERS;
-        queryName = 'getPrayers';
-        type = 'church';
+        queryTag = GET_PRAYERS;
+        label = 'My Church';
         break;
     }
-    return { query, queryName, type, variables };
+    return { queryTag, label, variables };
   };
 
   render() {
-    const { query, queryName, type, variables } = this.calculateQuery();
-    const { navigation } = this.props;
+    const { queryTag, label, variables } = this.calculateQuery();
 
     return (
-      <ModalView onClose={() => navigation.pop()}>
-        <PaddedFeedView>
+      <ModalView onClose={() => this.props.navigation.pop()}>
+        <FlexedSafeAreaView>
           <Query
-            query={query}
+            query={queryTag}
             variables={variables}
             fetchPolicy="cache-and-network"
           >
-            {({ data }) => {
-              const prayers = get(data, queryName, []);
-              return prayers.length > 0 ? (
-                <FlatList
-                  ref={(scroller) => {
-                    this.scroller = scroller;
+            {({ loading, data }) => {
+              // TODO: create a pretty loading state
+              if (loading) return null;
+              const prayers = get(data, this.props.query, []);
+              const prayer = prayers[this.state.prayerIndex];
+              const isLastPrayer =
+                this.state.prayerIndex + 1 === prayers.length;
+              return (
+                <Mutation
```

Comment:
I'd definitely love to figure out how to not have to nest these, but I don't off hand know a good way to do this. So 👍.

Reply:
@richarddubay Isn't nesting the Mutation within the Query pretty much standard practice though?

Reply:
@cdanwards Probably? It just feels weird to me, you know? But for sure, it's nothing we need to change now. I'm wondering if there is documentation somewhere?